sympy/polys/domains/finitering.py

```diff
@@ -0,0 +1,106 @@
+"""Implementation of :class:`FiniteRing` class. """
+
+from __future__ import print_function, division
+
+from sympy.polys.domains.field import Field
+from sympy.polys.domains.simpledomain import SimpleDomain
+from sympy.polys.domains.groundtypes import SymPyInteger
+from sympy.polys.domains.modularinteger import ModularIntegerFactory
+
+from sympy.polys.polyerrors import CoercionFailed
+from sympy.utilities import public
+
+
+@public
+class FiniteRing(Field, SimpleDomain):
+    """General class for finite rings. """
+
+    rep = 'FR'
+
+    is_FiniteRing = is_FR = True
+    is_Numerical = True
+
+    has_assoc_Ring = False
+    has_assoc_Field = True
+
+    dom = None
+    mod = None
+
+    def __init__(self, mod, dom=None, symmetric=True):
+        if mod <= 0:
+            raise ValueError('modulus must be a positive integer, got %s' % mod)
+        if dom is None:
+            from sympy.polys.domains import ZZ
+            dom = ZZ
+
+        self.dtype = ModularIntegerFactory(mod, dom, symmetric, self)
+        self.zero = self.dtype(0)
+        self.one = self.dtype(1)
+        self.dom = dom
+        self.mod = mod
+
+    def __str__(self):
+        return 'GF(%s)' % self.mod
```

Comment:
This should be changed.

Reply:
It should be changed to what? FR(%s) ?

Reply:
That is a possibility. It should be an expression that SymPy can correctly input to create a FiniteRing.

Reply:
Can you tell me how to go about this because I don't know where to start from.

Reply:
You could investigate how SymPy reads `FF(5)`, for example.

Reply:
I was trying to trace where and how FF is defined and hit a wall. 

polys/domains/finitefield.py has a declaration of rep = 'FF' in the code. 
polys/domains/__init__.py assigns FF as -
FF,ZZ,QQ = _GROUND_TYPES_MAP[GROUND_TYPES]

GROUND_TYPES is defined in core/compatibility.py but I can't seem to understand its working.

If 'FR' has to be defined for a FiniteRing will it also be declared in a similar manner?

Reply:
> will it also be declared in a similar manner?

I think so. It should probably be added to the list of ground types.